lib/charms/mysql/v0/mysql.py

```diff
@@ -975,6 +1020,38 @@ def remove_router_from_cluster_metadata(self, router_id: str) -> None:
             logger.exception(f"Failed to remove router from metadata with ID {router_id}")
             raise MySQLRemoveRouterFromMetadataError(e.message)
 
+    def set_dynamic_variable(
+        self,
+        variable: str,
+        value: str,
+        persist: bool = False,
+        instance_address: Optional[str] = None,
+    ) -> None:
+        """Set a dynamic variable value for the instance.
+
+        Args:
+            variable: The name of the variable to set
+            value: The value to set the variable to
+            persist: Whether to persist the variable value across restarts
+            instance_address: instance address to set the variable, default to current
+
+        Raises:
+            MySQLSetVariableError
+        """
+        if not instance_address:
+            instance_address = self.instance_address
+        logger.debug(f"Setting {variable} to {value} on {instance_address}")
+        set_var_command = [
+            f"shell.connect('{self.server_config_user}:{self.server_config_password}@{instance_address}')",
+            f"session.run_sql(\"SET {'PERSIST' if persist else 'GLOBAL'} {variable}={value}\")",
```

Comment:
Side comment: we must be careful here, to prevent SQL injection:
> value="something; drop database mysql;"

As I can see the method used in MySQL VM tests only... but who knows in the future.

Reply:
https://warthogs.atlassian.net/browse/DPE-2399